source/common/config/subscription_factory_impl.cc

```diff
@@ -105,5 +106,27 @@ SubscriptionPtr SubscriptionFactoryImpl::subscriptionFromConfigSource(
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+SubscriptionPtr SubscriptionFactoryImpl::collectionSubscriptionFromUrl(
+    const udpa::core::v1::ResourceLocator& collection_locator,
+    const envoy::config::core::v3::ConfigSource& /*config*/, absl::string_view /*type_url*/,
+    Stats::Scope& scope, SubscriptionCallbacks& callbacks,
+    OpaqueResourceDecoder& resource_decoder) {
+  std::unique_ptr<Subscription> result;
+  SubscriptionStats stats = Utility::generateStats(scope);
+
+  switch (collection_locator.scheme()) {
+  case udpa::core::v1::ResourceLocator::FILE: {
+    const std::string path = "/" + absl::StrJoin(collection_locator.id(), "/");
+    Utility::checkFilesystemSubscriptionBackingPath(path, api_);
+    return std::make_unique<Config::FilesystemCollectionSubscriptionImpl>(
+        dispatcher_, path, callbacks, resource_decoder, stats, validation_visitor_, api_);
+  }
+  default:
```

Comment:
Is this checked by schema? If not, maybe better to not use default and specify all of the options so we get a compile error if something is added?

Reply:
This is an enum, so you have weird values like `ResourceLocator_Scheme_ResourceLocator_Scheme_INT_MIN_SENTINEL_DO_NOT_USE_`